### ISSUE TITLE
Improve item visual recognition

### ISSUE THREAD

**Issue by Cwpute** [OxygenCobalt/Auxio#927]
# Description

Item readability could be greatly improved so that users know instantly what type of item they are looking, whether it is a track, an album, an artist or a playlist.

- [ ] 1. Change the default track cover art to a music note so it doesn't look like the album default cover art
- [ ] 2. Change default artist visual to circle-shaped art
- [ ] 3. Remove genre art, list them like tags

# Problem solved

In general: not knowing exactly which type of item we're looking at. For example, when using the global search bar, all the items are listed together (to be fair, in their own category). See screenshot below:
[![Screenshot-20241211-103533.png](https://i.postimg.cc/RhhV87Rw/Screenshot-20241211-103533.png)](https://postimg.cc/dZMYZTg0)
Otherwise, when switching tabs between albums, artists, playlists etc… you're looking at the list, not the tab names above. Although a quick look suffices, it's still needed sometimes. There is text accompanying each listed item that allows you to infer which category of item they belong in, but it's not explicit.

More specifically, and for each point i mentionned:

1. When browsing tracks and albums without covers, both have the same default cover art. The size difference between the two helps, as well as the categorization, but it's not sufficient to know it from the first look. 
2. Allows to easily discriminate between albums and artists, for artists who have an eponymous album, or a similar/duplicate art for both album and artist category. In addition, circle-shaped art for artists is quite standard as seen in "Other implementations" below.
3. Associating an image with a genre can be pretty misleading, and as your music folder gets larger and better tagged, it fuels your genres list with more niche or combined genres that are in turns associated with only a few covers, repeated in all your search list. Removing covers kills two birds with one stone: avoid that problem entirely, and differentiate genres from other items… with the added benefit of making a list of genres more compact.

# Other implementations

* Deezer (top to bottom: artists, titles, albums):
[![Screenshot-20241211-113959-1.png](https://i.postimg.cc/CKSrSmqT/Screenshot-20241211-113959-1.png)](https://postimg.cc/xXp3PG5t)

* Spotify (artists albums and tracks mixed in search results):
[![Screenshot-20241211-113635.png](https://i.postimg.cc/Wpg9FBwL/Screenshot-20241211-113635.png)](https://postimg.cc/CdMC962m)

* [Symphony player](https://github.com/zyrouge/symphony) (used to do this in previous versiwns, doesn't do it anymore):
[![screenshots.png](https://i.postimg.cc/W4qmRZ9L/screenshots.png)](https://postimg.cc/kV9RK23f)

# Benefit and shortcomings

It will further help users navigate the app, recognize the tab they are browsing, and recognize which items are which among each other without relying on in-app context. It will greatly help navigate search resluts as it will provide visual cues as to what they're looking at.

I don't have a fitting solution for playlist, but i figured the current implementation was good enough as playlists essentially work like user-created albums. Users can be expected to recognize their own playlists among other albums… although this may not be true for imported playlists and such.

I understand these changes may be quite hard to implement, for small sensible result to most current users, but it will definitely help people who are accustomed to online services such as Spotify which already do this kind of differenciation.
 
# Duplicates

- [x] I have checked this feature request for duplicates.
- [x] I have checked that this feature is not implemented in the lastest version.
- [x] I have checked the Why Are These Features Missing? page.
- [x] I agree to the Contribution Guidelines.

**Comment by OxygenCobalt** [comment]
I've been thinking of more cover presentations, specifically:'

- Artists: Circle cropping using artist images (#68) or a cascading cover collage
- Genre: A cluster of various album artworks (IIRC google photos does something like this)

it's difficult since I have a limited amount of time to do image transforms and coil sadly provides no good primitives for "many-image transforms".

Already doing a huge image loader refactor, will see if it can ride onto 4.0.0.

**Comment by Cwpute** [comment]
> Genre: A cluster of various album artworks (IIRC google photos does something like this)

Good idea, but it wouldn't work around the problem i pointed to : "as your music folder gets larger and better tagged, it fuels your genres list with more niche or combined genres that are in turns associated with only a few covers".
For playlists though, that would be quite good !

**Comment by OxygenCobalt** [comment]
>  "as your music folder gets larger and better tagged, it fuels your genres list with more niche or combined genres that are in turns associated with only a few covers".
For playlists though, that would be quite good !

My hope is that a genre's "cluster" would have an arrangement seeded from it's cover composition, so two genres with small cover samples will still have different clusters. Either way, I'll take in your ideas.

**Comment by slushspirit** [comment]
hi! would i be able to prioritize this? Thanks! <3 @OxygenCobalt

**Comment by OxygenCobalt** [comment]
Prioritized @slushspirit, this maxes out the prioritizations that you have as an $8 sponsor.

**Comment by OxygenCobalt** [comment]
Here is my rough idea behind the new image designs @slushspirit @Cwpute. Each one is deliberately going to be unique.

![Image](https://github.com/user-attachments/assets/be5bcd32-efe2-4a19-ba07-1ad9497af66d)

Note that the randomization is seeded based on how the artist/genre's UID. Therefore distinct genres/artists that might have very similar cover compositions will ideally appear differently and be identifiable.

**Comment by Cwpute** [comment]
Looks quite good ! excited to see it in action eventually

**Comment by slushspirit** [comment]
looks great! same, im excited to see it as well!

**Comment by OxygenCobalt** [comment]
4.1.0 @slushspirit @slushspirit @saintbyte 

![Screenshot_20260119-202901.png](https://github.com/user-attachments/assets/cf55dfd0-5a2e-4bec-adfa-dddc67dd4ef6)

**Comment by OxygenCobalt** [comment]
@Cwpute @slushspirit 

![Screenshot_20260124-095748.png](https://github.com/user-attachments/assets/baf3ae9f-0388-46ac-ba4e-ca493b37c22d)

![Screenshot_20260124-095740.png](https://github.com/user-attachments/assets/94d61a52-a04d-49fd-9093-4fa5835ec609)

![Screenshot_20260124-095722.png](https://github.com/user-attachments/assets/640485c3-1509-473e-b8c5-6abcbff3c627)

Really like how this turned out

**Comment by Cwpute** [comment]
Fantastic work ! Artist is realy great !
I still think either Playlist or Genre would need to have a different shape to easily be differentiated - in that regard your original design for Playlist would have made it stand out better.
But it's really good as is nonetheless !

**Comment by OxygenCobalt** [comment]
> diamond or oval

This isn't really standard and would probably look bad to be honest, sorry. I hope the visual arrangement is enough though @Cwpute

**Comment by Cwpute** [comment]
Yes, i changed my comment before i saw your answer, you're right.

**Comment by OxygenCobalt** [comment]
I do agree the stacking would be better but due to the nature of how android views are bounded I couldn't really pull it off w/o it looking odd (ex. smaller than other images) @Cwpute. Hope that helps.
